src/main/scala/it/pagopa/interop/agreementmanagement/model/persistence/AgreementPersistentBehavior.scala

```diff
@@ -56,6 +57,20 @@ object AgreementPersistentBehavior {
           )
           .getOrElse(handleFailure(AgreementNotFound(agreementId))(replyTo))
 
+      case AddAgreementContract(agreementId, contract, replyTo) =>
+        val addedDocument: Either[ComponentError, PersistentAgreementDocument] =
+          for {
+            agreement <- state.agreements.get(agreementId).toRight(AgreementNotFound(agreementId))
+            contract  <- agreement.contract.fold[Either[ComponentError, PersistentAgreementDocument]](Right(contract))(
+              _ => Left(AgreementDocumentAlreadyExists(agreementId))
+            )
+          } yield contract
```

Comment:
`contract` at row 64 shades `contract` at row 60, it's just because they have different types that the code compiles. How about renaming it to `result`?

Plus, if you like using cats you can reduce the verbosity of the 2nd statement in the for comprehension with something like 
```scala
agreement.contract.toLeft(contract).leftMap(_ => AgreementDocumentAlreadyExists(agreementId))
```
Otherwise, keeping the fold, you can use the `asLeft` and `asRight` helpers that differently from `Left` and `Right` constructor already widen the type to either. 
```scala
result  <- agreement.contract.fold(contract.asRight[ComponentError])(_ =>
    AgreementDocumentAlreadyExists(agreementId).asLeft[PersistentAgreementDocument]
)
```